sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorRedirectPolicy.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.opentelemetry.exporter;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.net.HttpURLConnection;
+
+/**
+ * A HttpPipeline policy that retries when a HTTP Redirect is received as response.
+ */
+public final class AzureMonitorRedirectPolicy implements HttpPipelinePolicy {
+
+    private static final int PERMANENT_REDIRECT_STATUS_CODE = 308;
+    private static final int MAX_REDIRECT_RETRIES = 10;
+    private final ClientLogger logger = new ClientLogger(AzureMonitorRedirectPolicy.class);
+    private String redirectedEndpointUrl;
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        return attemptRetry(context, next, context.getHttpRequest(), 0);
+    }
+
+    /**
+     *  Function to process through the HTTP Response received in the pipeline
+     *  and retry sending the request with new redirect url.
+     */
+    private Mono<HttpResponse> attemptRetry(final HttpPipelineCallContext context,
+                                            final HttpPipelineNextPolicy next,
+                                            final HttpRequest originalHttpRequest,
+                                            final int retryCount) {
+        // make sure the context is not modified during retry, except for the URL
+        context.setHttpRequest(originalHttpRequest.copy());
+        if (this.redirectedEndpointUrl != null) {
+            context.getHttpRequest().setUrl(this.redirectedEndpointUrl);
+        }
+        return next.clone().process()
+            .flatMap(httpResponse -> {
+                if (shouldRetryWithRedirect(httpResponse.getStatusCode(), retryCount)) {
+                    String responseLocation = httpResponse.getHeaderValue("Location");
```

Comment:
Will this return the full URL or just the hostname? Depending on what is returned, we may have to additionally append the path params.

Reply:
this will return the full URL